app/scripts/metamask-controller.js

```diff
@@ -6226,6 +6481,53 @@ export default class MetamaskController extends EventEmitter {
       origin = new URL(sender.url).origin;
     }
 
+    // solana account changed notifications
+    let caip25Caveat;
+    try {
+      caip25Caveat = this.permissionController.getCaveat(
+        origin,
+        Caip25EndowmentPermissionName,
+        Caip25CaveatType,
+      );
+    } catch {
+      // noop
+    }
+    if (caip25Caveat) {
+      const solanaAccountsChangedNotifications =
+        caip25Caveat.value.sessionProperties[
+          KnownSessionProperties.SolanaAccountChangedNotifications
+        ];
+
+      const sessionScopes = getSessionScopes(caip25Caveat.value, {
+        getNonEvmSupportedMethods: this.getNonEvmSupportedMethods.bind(this),
+      });
+
+      const solanaScope =
+        sessionScopes[MultichainNetworks.SOLANA] ||
+        sessionScopes[MultichainNetworks.SOLANA_DEVNET] ||
+        sessionScopes[MultichainNetworks.SOLANA_TESTNET];
+
+      if (solanaAccountsChangedNotifications && solanaScope) {
+        const { accounts } = solanaScope;
```

Comment:
the assumption here is that all solana accounts will be available on any solana scope.

The other way we could approach this would be to `const solanaAccounts = getPermittedAccountsForScopes(..., [solana_scopes,go,here])` and then if `solanaAccountsChangedNotifications && solanaAccounts.length !== 0`?..

I guess that wouldn't work if there are solana scopes, but no solana accounts. Do we want to emit accountChanged with empty array in that case still?